### ISSUE TITLE
Secure/HTTP cookies support

### ISSUE THREAD

**Issue by Alchemication** [Mohamed-512/Extra-Streamlit-Components#17]
Hi Mohamed,
We do have HTTP/Secure cookies, which are not read with `cookie_manager.get_all` API.
Sorry, not really familiar with Streamlit architecture, but basially are cookies are read from the front end JS? Meaning that obtaining an HTTP cookie is impossible?

**Comment by Mohamed-512** [comment]
If you mean getting a cookie from an HTTP/HTTPS request in the cookie header, then this is something the developer needs to take care of manually. By taking those cookies from the headers and using the CookieManager to set it. 

Streamlit is mainly a backend that serves a frontend and keeps communication with in a WebSocket format, which tells it to do specific things. Such as, add this new HTML tag, remove that specific row from a table, or -as in our current example- insert or delete cookies from your local browser storage -which is facilitated by the CookieManager-.

**Comment by Alchemication** [comment]
Yes, I mean getting a cookie from an HTTP/HTTPS request.
I absolutely have no idea how to get to that inside a Streamlit app.
Like you know in "standard" frameworks, we typically have a `request` object with headers, query params, json payload, etc. but would you have any idea how to do that in the context of a Streamlit app.
I don't mind to dig a bit in a JS code if needed.
I would say it would be a great addition to Streamlit eco-system, otherwise for us it's impossible to implement any type of authentication (we must use cookies to read user roles/data, etc.).

**Comment by Alchemication** [comment]
Yeah. That does not work from inside the Streamlit app. I do need cookies, not the headers.

And unfortunately when tried to use `requests` and then get the cookies from it - it only shows the cookies coming back from another service, not what's currently set in the Streamlit app.

Look, no worries, I will need to be more creative about this. We do have a single sign on system, and can't control how it works, all I have is a cookie with user info - but I can't read it from the Streamlit app right now ;(

**Comment by Alchemication** [comment]
Actually, scratch what I said above.

What I need is to somehow read the request/response headers that I see when any Streamlit resources are loaded. When I do `requests.get(google_url).headers` -- these headers do not come up.

I have to sleep on this I think ;-)
